### PR TITLE
Fixing Pause Demo - Object not rotating + 4.0 changes

### LINKED ISSUE #700: Pause Demo 4.0-dev not working
**Which demo project is affected:**
https://github.com/godotengine/godot-demo-projects/tree/4.0-dev/misc/pause

**OS/device including version:**
Arch Linux LTS-5.15.31-1 with i3 window manager

**Issue description:**
Cube doesn't rotate and errors keep happening

### PR DESCRIPTION
**Pull request by voylin**
Fixes #700

### CODE REVIEW

File: misc/pause/spinpause.tscn [hunk 1]
```diff
@@ -1,30 +1,28 @@
...
 
 [node name="PauseScene" type="Node3D"]
 
 [node name="Cube" type="MeshInstance3D" parent="."]
-mesh = SubResource( 1 )
-surface_material_override/0 = null
+mesh = SubResource( "1" )
 
 [node name="Camera3D" type="Camera3D" parent="."]
 transform = Transform3D(0.571594, 0.275303, -0.772974, 0, 0.942035, 0.335515, 0.820537, -0.191779, 0.538461, -5.59754, 2.75935, 4.01344)
```
Comment (aaronfranke) [inline misc/pause/spinpause.tscn:28]: The cube's default size has changed into 1 meter in diameter in Godot 4. As a result, we should move the camera and the light closer.

```suggestion
transform = Transform3D(0.707107, -0.241845, 0.664463, 0, 0.939693, 0.34202, -0.707107, -0.241845, 0.664463, 2, 1, 2)
```

File: misc/pause/spinpause.tscn [hunk 2]
```diff
@@ -33,7 +31,7 @@ near = 0.1
 
 [node name="AnimationPlayer" type="AnimationPlayer" parent="."]
 autoplay = "spin"
-anims/spin = SubResource( 2 )
+anims/spin = SubResource( "2" )
 
 [node name="SpotLight3D" type="SpotLight3D" parent="."]
 transform = Transform3D(0.792992, 0.251051, -0.555101, 0, 0.911149, 0.412078, 0.609232, -0.326775, 0.722534, -3.05357, 1.80053, 3.64099)
```
Comment (aaronfranke) [inline misc/pause/spinpause.tscn:37]: The cube's default size has changed into 1 meter in diameter in Godot 4. As a result, we should move the camera and the light closer.

```suggestion
transform = Transform3D(0.707107, -0.353554, 0.612373, 0, 0.866026, 0.5, -0.707107, -0.353554, 0.612373, 2, 2, 2)
```

For simplicity's sake, we should also reset the spotlight's range to the default value of 5 meters.

**Review by aaronfranke — COMMENTED** [review]
We should be setting demos to Vulkan Mobile if it doesn't look any different compared to Vulkan Clustered. Especially for GUI-only demos, they will look the same in Vulkan Mobile. Go to Project -> Project Settings -> General tab -> Rendering section -> Vulkan -> Back End -> Select Forward Mobile.

**Review by aaronfranke — COMMENTED** [review]
The `button_pressed` parameter in the script (`pause_button.gd` lines 11-15) needs to be renamed to `is_button_pressed` or similar.

**Comment by voylin** [comment]
@aaronfranke Thank you for the suggestion, everything is fixed now.

**Review by aaronfranke — APPROVED** [review]
Thanks!

**Pull request merged**